bodhi/server/services/updates.py

```diff
@@ -571,16 +571,16 @@ def new_update(request):
                 for update in updates:
                     # Validate that <koji_tag>-pending-signing and <koji-tag>-testing exists,
                     # if not create them
-                    side_tag_pending_signing = update.release.get_pending_signing_side_tag(
+                    side_tag_signing_pending = update.release.get_pending_signing_side_tag(
                         from_tag)
-                    side_tag_testing = update.release.get_testing_side_tag(from_tag)
-                    if not koji.getTag(side_tag_pending_signing):
-                        koji.createTag(side_tag_pending_signing, parent=from_tag)
-                    if not koji.getTag(side_tag_testing):
-                        koji.createTag(side_tag_testing, parent=from_tag)
-
-                    to_tag = side_tag_pending_signing
-                    # Move every new build to <from_tag>-pending-signing tag
+                    side_tag_testing_pending = update.release.get_testing_side_tag(from_tag)
+                    if not koji.getTag(side_tag_signing_pending):
+                        koji.createTag(side_tag_signing_pending, parent=from_tag)
+                    if not koji.getTag(side_tag_testing_pending):
+                        koji.createTag(side_tag_testing_pending, parent=from_tag, perm="autosign")
```

Comment:
the CreateTag api from koji expect a perm_id (integer) and not a permission name. The only API that seems to allow to pass a permission name is editTag2.

```suggestion
                        koji.createTag(side_tag_testing_pending, parent=from_tag)
                        koji.editTag2(side_tag_testing_pending, perm="autosign")
```

Reply:
https://github.com/fedora-infra/bodhi/pull/3631/files#diff-14b55177c4dbffd842bef13afbea8a2aL506-R507
looked like it allowed for names

Reply:
This is what I get trying it in stg

```
>>> koji.createTag("f32-build-side-7861-pending-signing", parent="f32-build-side-7861", perm="autosign")
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/usr/lib/python3.7/site-packages/koji/__init__.py", line 2090, in __call__
    return self.__func(self.__name, args, opts)
  File "/usr/lib/python3.7/site-packages/koji/__init__.py", line 2594, in _callMethod
    raise err
  File "/usr/lib/python3.7/site-packages/koji/__init__.py", line 2575, in _callMethod
    return self._sendCall(handler, headers, request)
  File "/usr/lib/python3.7/site-packages/koji/__init__.py", line 2497, in _sendCall
    return self._sendOneCall(handler, headers, request)
  File "/usr/lib/python3.7/site-packages/koji/__init__.py", line 2539, in _sendOneCall
    ret = self._read_xmlrpc_response(r)
  File "/usr/lib/python3.7/site-packages/koji/__init__.py", line 2551, in _read_xmlrpc_response
    result = u.close()
  File "/usr/lib64/python3.7/xmlrpc/client.py", line 656, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault 1: '<class \'psycopg2.DataError\'>: invalid input syntax for integer: "autosign"\nLINE 1: ...) VALUES (\'\', 41210584, 428, false, false, false, \'autosign\'...\n                                                             ^\n'>
```

Reply:
> https://github.com/fedora-infra/bodhi/pull/3631/files#diff-14b55177c4dbffd842bef13afbea8a2aL506-R507
> looked like it allowed for names

Ok this is the dummy buildsystem that you linked here, so indeed we can accept anything here :-)